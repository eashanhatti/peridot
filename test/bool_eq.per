/* Typechecks */
define p: Equal(true, true) = reflexive

/* Does not typecheck */
define q: Equal(true, false) = reflexive
