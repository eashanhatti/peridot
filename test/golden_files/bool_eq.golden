((fromList
    [(UVGlobal 1000, Just (Rigid (TwoType))),
     (UVGlobal 1001, Just (Rigid (TwoType))),
     (UVGlobal 1002, Just (Rigid (TwoIntro0))),
     (UVGlobal 1003, Just (Rigid (TwoType))),
     (UVGlobal 1004, Just (Rigid (TwoType))),
     (UVGlobal 1005, Nothing),
     (UVGlobal 1006, Just (Rigid (TwoType))),
     (UVGlobal 1007, Just (Rigid (TwoType))),
     (UVGlobal 1008, Just (Rigid (TwoType))),
     (UVGlobal 1009, Just (Rigid (TwoType)))],
  fromList [],
  fromList
    [(SourcePos
        {sourceName = "<TODO>",
         sourceLine = Pos 5,
         sourceColumn = Pos 32},
      FailedUnify
        (Rigid
           (ObjIdType (Rigid (TwoIntro0)) (Rigid (TwoIntro1))))
        (Rigid
           (ObjIdType
              (Neutral _ (UniVar (UVGlobal 1005)))
              (Neutral _ (UniVar (UVGlobal 1005))))))]),
 Declare
   Obj
   (Rigid (NameIntro Obj (Id {unId = 0})))
   (Rigid
      (ObjIdType (Rigid TwoIntro0) (Rigid TwoIntro0)))
   (Declare
      Obj
      (Rigid (NameIntro Obj (Id {unId = 1})))
      (Rigid
         (ObjIdType (Rigid TwoIntro0) (Rigid TwoIntro1)))
      (Define
         (Rigid (NameIntro Obj (Id {unId = 0})))
         (Rigid (ObjIdIntro (UniVar (UVGlobal 1002))))
         (Define
            (Rigid (NameIntro Obj (Id {unId = 1})))
            (Rigid (ObjIdIntro (UniVar (UVGlobal 1005))))
            (Rigid (TypeType Obj))))))
